validator/client/validator_test.go

```diff
@@ -287,16 +293,73 @@ func TestCanonicalHeadSlot_OK(t *testing.T) {
 		t.Errorf("Mismatch slots, wanted: %v, received: %v", params.BeaconConfig().GenesisSlot, headSlot)
 	}
 }
+func TestWaitMultipleActivation_LogsActivationEpochOK(t *testing.T) {
+	hook := logTest.NewGlobal()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	client := internal.NewMockValidatorServiceClient(ctrl)
+
+	v := validator{
+		keys:            keyMapThreeValidators,
+		validatorClient: client,
+	}
+	clientStream := internal.NewMockValidatorService_WaitForActivationClient(ctrl)
+	client.EXPECT().WaitForActivation(
+		gomock.Any(),
+		&pb.ValidatorActivationRequest{
+			PublicKeys: publicKeys(v.keys),
+		},
+	).Return(clientStream, nil)
+	clientStream.EXPECT().Recv().Return(
+		&pb.ValidatorActivationResponse{
+			ActivatedPublicKeys: publicKeys(v.keys),
+		},
+		nil,
+	)
+	if err := v.WaitForActivation(context.Background()); err != nil {
+		t.Errorf("Could not wait for activation: %v", err)
+	}
+	testutil.AssertLogsContain(t, hook, "Validator activated")
+}
+
+// func TestUpdateAssignments_DoesNothingWhenNotEpochStartAndAlreadyExistingAssignments(t *testing.T) {
+// 	ctrl := gomock.NewController(t)
+// 	defer ctrl.Finish()
+// 	client := internal.NewMockValidatorServiceClient(ctrl)
+
+// 	slot := uint64(1)
+// 	v := validator{
+// 		keys:            keyMap,
+// 		validatorClient: client,
+// 		assignments: &pb.CommitteeAssignmentResponse{
+// 			Assignment: []*pb.CommitteeAssignmentResponse_CommitteeAssignment{
+// 				&pb.CommitteeAssignmentResponse_CommitteeAssignment{
+// 					Committee: []uint64{},
+// 					Slot:      10,
+// 					Shard:     20,
+// 				},
+// 			},
+// 		},
+// 	}
+// 	client.EXPECT().CommitteeAssignment(
+// 		gomock.Any(),
+// 		gomock.Any(),
+// 	).Times(0)
+
+// 	if err := v.UpdateAssignments(context.Background(), slot); err != nil {
+// 		t.Errorf("Could not update assignments: %v", err)
+// 	}
```

Comment:
Why is this commented out? Please fix

Reply:
test fails because of && false

https://github.com/shayzluf/prysm/blob/20717ee7528172fa8e99df0532be71219ad01fb1/validator/client/validator.go#L144-L148

Reply:
It isn't failing in master, why does it fail now?

Reply:
what should prevent it from calling update when it gets pass 
```// Testing run time for fetching every slot. This is not meant for production!
	if slot%params.BeaconConfig().SlotsPerEpoch != 0 && v.assignments != nil && false {
		// Do nothing if not epoch start AND assignments already exist.
		return nil
	}
```
then its not doing nothing

Reply:
found the answer on main branch t.Skip()